### ISSUE TITLE
NotImplementedError from Money#hashCode

### ISSUE THREAD

**Issue by psttf** [typelevel/squants#323]
with Squants 1.4.0:

```scala
Welcome to Scala 2.12.7 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_181).
Type in expressions for evaluation. Or try :help.

scala> import squants.market._
import squants.market._

scala> Money(123.456, USD)
res0: squants.market.Money = 123.456 USD

scala> res0.hashCode
scala.NotImplementedError: an implementation is missing
  at scala.Predef$.$qmark$qmark$qmark(Predef.scala:284)
  at squants.market.Money$.primaryUnit(Money.scala:404)
  at squants.market.Money$.primaryUnit(Money.scala:385)
  at squants.Quantity.hashCode(Quantity.scala:182)
  ... 36 elided
```

**Comment by hunterpayne** [comment]
This is a real bug.  It is caused by the fact that hashCode comes from the Quantity class.  That implementation uses the hashCode of the Dimension and specifically the hashCode of the primaryUnit.  Because the Money class uses the MoneyContext for the primaryUnit, its lacks an implementation (???) to ensure the MoneyContext implementations are being used.  So the Money class needs its own hashCode implementation which is quite easy.  Here is one possible implementation:

  override def hashCode() = java.util.Objects.hash(currency, amount)

**Comment by garyKeorkunian** [comment]
It looks like this was corrected in #330.  @psttf Can you please verify?

**Comment by Vrolijkx** [comment]
Should be resolved on the latest master. So think we can close this one.

**Comment by garyKeorkunian** [comment]
Resolved by PR #351
